src/index.ts

```diff
@@ -128,6 +146,7 @@ export class Spinner {
   stop() {
     this.clearMultiLineOutput();
     this.end(false);
+    this.clearListeners();
```

Comment:
I think this should be called in the end method, since that is the definitive handler for stopping spinning whether the spinner is being hidden or changed to a finished state.

Reply:
You're right 👍 Will update it